crates/ra_ide_api/src/display/navigation_target.rs

```diff
@@ -290,83 +407,54 @@ impl NavigationTarget {
+        node.doc_comment_text()
     }
 
-    /// Get a description of this node.
-    ///
-    /// e.g. `struct Name`, `enum Name`, `fn Name`
-    pub(crate) fn description(&self, db: &RootDatabase) -> Option<String> {
-        // FIXME: After type inference is done, add type information to improve the output
-        let node = self.node(db)?;
-
-        fn visit_ascribed_node<T>(node: &T, prefix: &str) -> Option<String>
-        where
-            T: NameOwner + VisibilityOwner + TypeAscriptionOwner,
-        {
-            let mut string = visit_node(node, prefix)?;
-
-            if let Some(type_ref) = node.ascribed_type() {
-                string.push_str(": ");
-                type_ref.syntax().text().push_to(&mut string);
-            }
-
-            Some(string)
-        }
-
-        fn visit_node<T>(node: &T, label: &str) -> Option<String>
-        where
-            T: NameOwner + VisibilityOwner,
-        {
-            let mut string =
-                node.visibility().map(|v| format!("{} ", v.syntax().text())).unwrap_or_default();
-            string.push_str(label);
-            string.push_str(node.name()?.text().as_str());
-            Some(string)
-        }
+    visitor()
+        .visit(doc_comments::<ast::FnDef>)
+        .visit(doc_comments::<ast::StructDef>)
+        .visit(doc_comments::<ast::EnumDef>)
+        .visit(doc_comments::<ast::TraitDef>)
+        .visit(doc_comments::<ast::Module>)
+        .visit(doc_comments::<ast::TypeAliasDef>)
+        .visit(doc_comments::<ast::ConstDef>)
+        .visit(doc_comments::<ast::StaticDef>)
+        .visit(doc_comments::<ast::NamedFieldDef>)
+        .visit(doc_comments::<ast::EnumVariant>)
+        .visit(doc_comments::<ast::MacroCall>)
+        .accept(&node)?
+}
 
-        visitor()
-            .visit(|node: &ast::FnDef| Some(crate::display::function_label(node)))
-            .visit(|node: &ast::StructDef| visit_node(node, "struct "))
-            .visit(|node: &ast::EnumDef| visit_node(node, "enum "))
-            .visit(|node: &ast::TraitDef| visit_node(node, "trait "))
-            .visit(|node: &ast::Module| visit_node(node, "mod "))
-            .visit(|node: &ast::TypeAliasDef| visit_node(node, "type "))
-            .visit(|node: &ast::ConstDef| visit_ascribed_node(node, "const "))
-            .visit(|node: &ast::StaticDef| visit_ascribed_node(node, "static "))
-            .visit(|node: &ast::NamedFieldDef| visit_ascribed_node(node, ""))
-            .visit(|node: &ast::EnumVariant| Some(node.name()?.text().to_string()))
-            .accept(&node)?
-    }
+/// Get a description of a symbol.
+///
+/// e.g. `struct Name`, `enum Name`, `fn Name`
+fn description_from_symbol(db: &RootDatabase, symbol: &FileSymbol) -> Option<String> {
+    let file = db.parse(symbol.file_id).tree;
+    let node = symbol.ptr.to_node(file.syntax()).to_owned();
+
+    // FIXME: After type inference is done, add type information to improve the output
```

Comment:
this fixme is obsolete I think and can be removed

Reply:
at lest, it doesn't belong *here*, as symbols can't have type information associaed with them, because symbols are not hir, and just an approximate heuristic.

Reply:
Removed.